Common/Final.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Common
+{
+	/// <summary>
+	/// This class enforces one-time only initialization.
+	/// Normally you want to use `readonly` for that, but sometimes you can't
```

Comment:
In tests I sometimes hack things with reflection, get-only properties and readonly field are settable then.